src/rust/plugin-work-queue/src/psql_queue.rs

```diff
@@ -0,0 +1,305 @@
+use sqlx::{
+    Pool,
+    Postgres,
+};
+use tracing::instrument;
+use uuid::Uuid;
+
+#[derive(sqlx::Type, Debug, Clone, PartialEq, Eq)]
+#[sqlx(type_name = "status_t", rename_all = "lowercase")]
+pub enum Status {
+    Enqueued,
+    Failed,
+    Processed,
+}
+
+#[derive(Copy, Clone, Debug, sqlx::Type)]
+#[sqlx(transparent)]
+pub struct ExecutionId(i64);
+
+impl From<i64> for ExecutionId {
+    fn from(id: i64) -> Self {
+        Self(id)
+    }
+}
+
+impl From<ExecutionId> for i64 {
+    fn from(id: ExecutionId) -> Self {
+        id.0
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct NextExecutionRequest {
+    pub execution_key: ExecutionId,
+    pub plugin_id: uuid::Uuid,
+    pub tenant_id: uuid::Uuid,
+    pub pipeline_message: Vec<u8>,
```

Comment:
They can be pub(crate), but yes it's just a data object so that we can get sqlx to deserialize a given row for a specific query in a way that's ergonomic.

Reply:
Actually they have to be pub as the fields are accessed in tests.